### PR TITLE
fix: primary mentor cannot add himself again

### PR DESCRIPTION
**Pull request by priyam-03**
Fixes [#4541](https://github.com/CircuitVerse/CircuitVerse/issues/4541) 

#### Describe the changes you have made in this PR -
This pull request addresses the issue https://github.com/CircuitVerse/CircuitVerse/issues/4541 by modifying the group member controller. Here issue was primary mentor could add again himself as a mentor. So here if there is the mail of the primary mentor in request it is discarded. And also relevant notification is shown.

Changes Made:

- Added logic to discard primary mentor to add himself again
- Resolved the issue #4541 

### Screenshots of the changes (If any) -

https://github.com/CircuitVerse/CircuitVerse/assets/104610119/914ba128-4045-4177-b581-c566a89889e7

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: app/controllers/group_members_controller.rb [hunk 1]
```diff
@@ -38,8 +38,7 @@ def create
     group_member_emails = group_member_params[:emails].grep(Devise.email_regexp)
 
     present_members = User.where(id: @group.group_members.pluck(:user_id)).pluck(:email)
-    newly_added = group_member_emails - present_members
-
+    newly_added = group_member_emails - present_members - [current_user&.email]
```
Comment (tachyons) [inline app/controllers/group_members_controller.rb:41]:
```suggestion
    newly_added = group_member_emails - present_members - [current_user.email]
```

**Comment by tachyons** [comment]
Please add rspec tests for this

**Comment by priyam-03** [comment]
@tachyons I have added rspec tests
image

**Review by tanmoysrt — APPROVED** [review]
Working as expected :+1:

**Pull request merged**